README.md

```diff
@@ -48,95 +100,375 @@ Options:
+
+Presently, the accepted values for the `exists` keyword encompass:
+
+`optional`: Denotes that the NeXus concept is not mandatory and may be absent.
+`recommended`: Suggests that the NeXus concept is advisable but not mandatory.
+`required`: Indicates that the NeXus concet must be present within the structure.
+`[min, <number>, max, <number> or infty]`: Represents an array type value that signifies the multiplicity of the NeXus concepts. For instance, concept belonging `exists: [min, 3, max, infty]` implies that the concept must come a minimum of three instances and may extend to any number.
+
+**`exists` in YAML**
+
+```yaml
+transmission_correction(NXcalibration):
+  exists: optional
+  doc: |
+    This calibration procedure is used to account for the different tranmsission efficiencies.
+```
+In the above example the greoup `transmission_correction` is a optional group.
+
+### Keyword `unit`
+A statement introducing NeXus-compliant NXDL attribute `units` attributes to the `field` e.g. `NX_VOLTAGE` to assign predefied pyhsical unit.
+
+**`unit` in YAML**
+
+```yaml
+detector_voltage(NX_FLOAT):
+  unit: NX_VOLTAGE
+  doc: |
+    Voltage applied to detector.
+```
+
+### Keyword `dimensions`
+The `dimensions` term  describes the multidimensional nature of the data, specifying its rank, dimensional indices, and corresponding length of the rank. For example, the attribute `rank` defines the dimension of the data set. To elucidate each dimension we use other two keywords `dim` and `dim_parameters`. The `dim` keyword comprises an array of arrays, the nested array encapsulates values for `index` and `value` (NeXus keywords) pairs. Each array within the `dim` array corresponds to a specific dimension of the multidimensional data. For example, for 2D particle motion, the `dim` array may be represented as `[[0, nx], [1, ny]]`, indicating the axes index and its length. Another keyword `dim_parameters` contains further information of each dimension such as `doc`, `ref`, etc. It is important to note that each term or keyword within `dim_parameters` must have the same length as the value of the rank keyword.
+
+**`dimensions` in YAML**
+```yaml
+# 2D particle motion
+dimensions:
+   rank: 2
+   dim: [[0, nx], [1, ny]]
+   dim_parameters:
+      doc: ["Position of particle on x-axis.","Position of particle on y-axis."]
+```
+The `dimensions` can also be written in shorter form
+**Dimensions in YAML (shorter form)**
+```yaml
+# 2D particle motion
+dimensions:
+   rank: 2
+   dim: (nx, ny)
+```
+
+### Keyword `enumeration`
+Python list of strings which are considered as recommended items for the fields or attributes.
```

Comment:
```suggestion
Python-like list of strings which are considered as recommended items for the fields or attributes.
```

It is not just recommended items, but rather you need to pick one of the enumeration items, so basically you are shrinking the space of possible values.